src/go/agwd/config/config_test.go

```diff
@@ -132,6 +132,53 @@ func TestParseTarget(t *testing.T) {
 	}
 }
 
+func TestTargetSchemeAdapter(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		input resolver.Target
+		want  resolver.Target
+	}{
+		{},
+		{
+			input: resolver.Target{
+				Scheme: "unix",
+			},
+			want: resolver.Target{
+				Scheme: "unix",
+			},
+		},
+		{
+			input: resolver.Target{
+				Scheme: "ipv4",
+			},
+			want: resolver.Target{
+				Scheme: "tcp4",
+			},
+		},
+		{
+			input: resolver.Target{
+				Scheme: "ipv6",
+			},
+			want: resolver.Target{
+				Scheme: "tcp6",
+			},
+		},
```

Comment:
probably good to add one more test that scheme with tcp4 outputs scheme with tcp4